### PR TITLE
Stringify gpg completions

### PR DESCRIPTION
**Pull request by ammgws**
## Description

Use builtin string function instead of sed.

P.S.: Is it possible to one-liner this using `string`?

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

### CODE REVIEW

File: share/functions/__fish_complete_gpg_key_id.fish [hunk 1]
```diff
@@ -5,13 +5,15 @@ function __fish_complete_gpg_key_id -d 'Complete using gpg key ids' -a __fish_co
     set -l keyid
     $__fish_complete_gpg_command --list-keys --with-colons | while read garbage
         switch $garbage
-            # Extract user ids
+            # Extract user ids (note: gpg escapes colons as '\x3a')
             case "uid*"
-                echo $garbage | cut -d ":" -f 10 | sed -e "s/\\\x3a/:/g" | read uid
+                set -l __uid (echo $garbage | string split ":")
```
Comment (faho) [inline share/functions/__fish_complete_gpg_key_id.fish]: String doesn't need to read from stdin. Use `string split : -- $garbage`.

**Comment by faho** [comment]
>P.S.: Is it possible to one-liner this using string?

Technically anything's possible with regex. You can match something like `([^:]*:){10}` to get the tenth field (or possibly the eleventh, I've not tried this), and that replaces that `cut`.

Easily? No, we lack the field selection for `string split`. I could swear there's an issue open about it, but I can't find it. It would be useful.

File: share/functions/__fish_complete_gpg_key_id.fish (hunk 1 above)
Comment (ammgws) [reply share/functions/__fish_complete_gpg_key_id.fish]: Oopsie. Cheers!

**Comment by ammgws** [comment]
> Easily? No, we lack the field selection for `string split`. I could swear there's an issue open about it, but I can't find it. It would be useful.

I guess #243?

**Comment by faho** [comment]
Yeah, that works. Merged, thanks!

**Comment by ohk2kt3t4** [comment]
> Easily? No, we lack the field selection for `string split`. I could swear there's an issue open about it, but I can't find it. It would be useful.

However at least this seems to work in my terminal:

`set uid (string replace -a '\x3a' ':' -- (string split ":" -- $garbage)[10])`

Am I overlooking something?

**Pull request merged**